domains/service/src/lib.rs

```diff
@@ -24,3 +47,336 @@ pub struct DomainConfiguration<AccountId> {
+    )?;
+    let sync_service_import_queue = sync_service.clone();
+    let sync_service = Arc::new(sync_service);
+
+    let genesis_hash = client
+        .hash(Zero::zero())
+        .ok()
+        .flatten()
+        .expect("Genesis block exists; qed");
+    let mut network_params = sc_network::config::Params::<TBl> {
+        role: config.role.clone(),
+        executor: {
+            let spawn_handle = Clone::clone(&spawn_handle);
+            Box::new(move |fut| {
+                spawn_handle.spawn("libp2p-node", Some("networking"), fut);
+            })
+        },
+        network_config: config.network.clone(),
+        genesis_hash,
+        protocol_id: protocol_id.clone(),
+        fork_id: config.chain_spec.fork_id().map(ToOwned::to_owned),
+        metrics_registry: config
+            .prometheus_config
+            .as_ref()
+            .map(|config| config.registry.clone()),
+        block_announce_config,
+        tx,
+        request_response_protocol_configs: vec![
+            Some(block_request_protocol_config),
+            Some(state_request_protocol_config),
+            warp_sync_protocol_config,
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<Vec<_>>(),
+    };
+
+    // crate transactions protocol and add it to the list of supported protocols of `network_params`
+    let transactions_handler_proto = sc_network_transactions::TransactionsHandlerPrototype::new(
+        protocol_id,
+        client
+            .block_hash(0u32.into())
+            .ok()
+            .flatten()
+            .expect("Genesis block exists; qed"),
+        config.chain_spec.fork_id(),
+    );
+    network_params
+        .network_config
+        .extra_sets
+        .insert(0, transactions_handler_proto.set_config());
+
+    let has_bootnodes = !network_params.network_config.boot_nodes.is_empty();
+    let network_mut = sc_network::NetworkWorker::new(network_params)?;
+    let network = network_mut.service().clone();
+
+    let (tx_handler, tx_handler_controller) = transactions_handler_proto.build(
+        network.clone(),
+        sync_service.clone(),
+        Arc::new(TransactionPoolAdapter::new(
+            transaction_pool,
+            client.clone(),
+        )),
+        config
+            .prometheus_config
+            .as_ref()
+            .map(|config| &config.registry),
+    )?;
+
+    spawn_handle.spawn(
+        "network-transactions-handler",
+        Some("networking"),
+        tx_handler.run(),
+    );
+    spawn_handle.spawn_blocking(
+        "chain-sync-network-service-provider",
+        Some("networking"),
+        chain_sync_network_provider.run(network.clone()),
+    );
+    spawn_handle.spawn(
+        "import-queue",
+        None,
+        import_queue.run(Box::new(sync_service_import_queue)),
+    );
+    spawn_handle.spawn_blocking("syncing", None, engine.run());
+
+    let (system_rpc_tx, system_rpc_rx) = tracing_unbounded("mpsc_system_rpc", 10_000);
```

Comment:
shouldn't we handle some of the rpc requests if not all ? I feel some of them are useful to have

Reply:
Sorry I did not quite follow here, it seems like this part is the same as `sc_service::build_network`.

Reply:
ahh nevermind. I was confused with this rpc future instead - https://github.com/paritytech/substrate/blob/master/client/service/src/lib.rs#L214